File: testdata/bench/kotlin/BenchmarkTest00005.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Vulnerable: SQL injection via string template with ORDER BY
suspend fun handler00005(call: ApplicationCall) {
    val sort = call.parameters["sort"]
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val rs = conn.createStatement().executeQuery("SELECT * FROM items ORDER BY $sort")
    call.respondText(rs.toString())
}

File: testdata/bench/go/BenchmarkTest00005.go
package bench

import (
	"database/sql"
	"net/http"
)

// SQL injection via goroutine
func Handler00005(w http.ResponseWriter, r *http.Request) {
	param := r.URL.Query().Get("sort")
	db, _ := sql.Open("sqlite3", ":memory:")
	go func() {
		query := "SELECT * FROM items ORDER BY " + param
		db.Query(query)
	}()
}
